File: voiceping-sdk/src/main/java/com/smartwalkie/voicepingsdk/NetworkUtil.java
package com.smartwalkie.voicepingsdk;

import android.content.Context;
import android.net.ConnectivityManager;
import android.net.NetworkInfo;


class NetworkUtil {

    static boolean isNetworkConnected(Context context) {
        ConnectivityManager connectivityManager = (ConnectivityManager) context.getSystemService(
                Context.CONNECTIVITY_SERVICE);
        NetworkInfo networkInfo = connectivityManager.getActiveNetworkInfo();
        return networkInfo != null && networkInfo.isConnectedOrConnecting();
    }
}

File: voiceping-sdk/src/main/java/com/smartwalkie/voicepingsdk/NetworkUtil.kt
package com.smartwalkie.voicepingsdk

import android.content.Context
import android.net.ConnectivityManager
import android.net.NetworkInfo

internal object NetworkUtil {

    @JvmStatic
    fun isNetworkConnected(context: Context): Boolean {
        val connectivityManager = context.getSystemService(
            Context.CONNECTIVITY_SERVICE
        ) as ConnectivityManager
        val networkInfo = connectivityManager.activeNetworkInfo
        return networkInfo != null && networkInfo.isConnectedOrConnecting
    }
}